core/vibes/soul/primitives/banner/index.tsx

```diff
@@ -0,0 +1,69 @@
+'use client';
+
+import { clsx } from 'clsx';
+import { X } from 'lucide-react';
+import { ForwardedRef, forwardRef, ReactNode, useCallback, useEffect, useState } from 'react';
+
+export const Banner = forwardRef(
+  (
+    {
+      children,
+      hideDismiss = false,
+      className,
+      onDismiss,
+    }: {
+      children: ReactNode;
+      hideDismiss?: boolean;
+      className?: string;
+      onDismiss?(): void;
+    },
+    ref: ForwardedRef<HTMLDivElement>,
+  ) => {
+    const [banner, setBanner] = useState({ dismissed: false, initialized: false });
+
+    useEffect(() => {
+      const hidden = localStorage.getItem('hidden-banner') === 'true';
+
+      setBanner({ dismissed: hidden, initialized: true });
+    }, []);
+
+    const hideBanner = useCallback(() => {
+      setBanner((prev) => ({ ...prev, dismissed: true }));
+      localStorage.setItem('hidden-banner', 'true');
```

Comment:
🍹 Should these banners have a unique id? That way if a merchants starts a new promotion the banner will show up again?

Reply:
Yeah, they should. And the ID should be a prop so that if using Makeswift, for example, the ID can be changed to have the promotion show up again. I think we probably need a PR in VIBES that fixes this